src/System.Runtime.Extensions/tests/System/AppDomainTests.cs

```diff
@@ -609,6 +610,59 @@ public void AssemblyResolve_RequestingAssembly()
         }
 
         [Fact]
+        [ActiveIssue("https://github.com/dotnet/corefx/issues/22555", TargetFrameworkMonikers.Uap)]
+        public void AssemblyResolve_IsNotCalledForCoreLibResources()
+        {
+            bool assemblyResolveHandlerCalled = false;
+            AppDomain.CurrentDomain.AssemblyResolve +=
+                (sender, e) =>
+                {
+                    // This implementation violates the contract. AssemblyResolve event handler is supposed to return an assembly
+                    // that matches the requested identity and that is not the case here.
+                    assemblyResolveHandlerCalled = true;
+                    return typeof(AppDomainTests).Assembly;
+                };
+
+            CultureInfo previousUICulture = CultureInfo.CurrentUICulture;
+            CultureInfo.CurrentUICulture = new CultureInfo("de-CH");
+            try
+            {
+                // The resource lookup for NullReferenceException (generally for CoreLib resources) should not raise the
+                // AssemblyResolve event because a misbehaving handler could cause an infinite recursion check and fail-fast to
+                // be triggered when the resource is not found, as the issue would repeat when reporting that error.
+                Assert.Throws<NullReferenceException>(() => ((string)null).Contains("a"));
+                Assert.False(assemblyResolveHandlerCalled);
+            }
+            finally
+            {
+                CultureInfo.CurrentUICulture = previousUICulture;
+            }
+        }
+
+        [Fact]
+        [ActiveIssue("https://github.com/dotnet/corefx/issues/22555", TargetFrameworkMonikers.Uap)]
+        public void AssemblyResolve_LoadFromHandlerChecksForNullRequestingAssembly()
+        {
+            Assert.ThrowsAny<Exception>(() => Assembly.LoadFrom(@"nonexistent.dll"));
```

Comment:
LoadFrom registers the AssemblyResolve event handler, and runs up to the last line where it throws FileNotFoundException. The handler gets called later and throws ArgumentNullException because the requesting assembly is null.